modules/siddhi-extensions/event-table/src/test/java/org/wso2/siddhi/extension/eventtable/hazelcast/DefineTableTestCase.java

```diff
@@ -0,0 +1,471 @@
+
+        SiddhiManager siddhiManager = new SiddhiManager();
+        TableDefinition tableDefinition = TableDefinition.id("cseEventStream").annotation(Annotation.annotation("from").element("eventtable", "hazelcast")).attribute("symbol", Attribute.Type.STRING).attribute("price", Attribute.Type.INT);
+        ExecutionPlan executionPlan = new ExecutionPlan("ep1");
+        executionPlan.defineTable(tableDefinition);
+        ExecutionPlanRuntime executionPlanRuntime = siddhiManager.createExecutionPlanRuntime(executionPlan);
+        List<String> hciNames = new ArrayList<String>();
+        for (HazelcastInstance hci : Hazelcast.getAllHazelcastInstances()) {
+            hciNames.add(hci.getName());
+        }
+        Assert.assertTrue(hciNames.contains(HazelcastEventTableConstants.HAZELCAST_INSTANCE_PREFIX + executionPlanRuntime.getName()));
+        executionPlanRuntime.shutdown();
+    }
+
+    @Test
+    public void testQuery2() throws InterruptedException {
+        log.info("testTableDefinition2 - OUT 0");
+
+        SiddhiManager siddhiManager = new SiddhiManager();
+        StreamDefinition streamDefinition = StreamDefinition.id("StockStream").attribute("symbol", Attribute.Type.STRING).attribute("price", Attribute.Type.INT);
+        TableDefinition tableDefinition = TableDefinition.id("StockTable").annotation(Annotation.annotation("from").element("eventtable", "hazelcast")).attribute("symbol", Attribute.Type.STRING).attribute("price", Attribute.Type.INT);
+        Query query = Query.query();
+        query.from(InputStream.stream("StockStream"));
+        query.insertInto("StockTable");
+        ExecutionPlan executionPlan = new ExecutionPlan("ep1");
+        executionPlan.addQuery(query);
+        executionPlan.defineStream(streamDefinition);
+        executionPlan.defineTable(tableDefinition);
+        ExecutionPlanRuntime executionPlanRuntime = siddhiManager.createExecutionPlanRuntime(executionPlan);
+
+        InputHandler stockStream = executionPlanRuntime.getInputHandler("StockStream");
+        executionPlanRuntime.start();
+        stockStream.send(new Object[]{"WSO2", 55.6f});
+        stockStream.send(new Object[]{"IBM", 75.6f});
+        Thread.sleep(500);
+
+        Map<String, HazelcastInstance> instanceMap = new HashMap<String, HazelcastInstance>();
+        for (HazelcastInstance hci : Hazelcast.getAllHazelcastInstances()) {
+            instanceMap.put(hci.getName(), hci);
+        }
+        Assert.assertTrue(instanceMap.containsKey(HazelcastEventTableConstants.HAZELCAST_INSTANCE_PREFIX + executionPlanRuntime.getName()));
+        HazelcastInstance instance = instanceMap.get(HazelcastEventTableConstants.HAZELCAST_INSTANCE_PREFIX + executionPlanRuntime.getName());
+        List<StreamEvent> streamEvents = instance.getList(HazelcastEventTableConstants.HAZELCAST_LIST_INSTANCE_PREFIX + executionPlanRuntime.getName() + '_' + tableDefinition.getId());
+        Assert.assertEquals(2, streamEvents.size());
```

Comment:
Since 'isEventsMatch' utility method is available, can we check whether the list has the correct events as well in addition to checking whether the number of events are correct?